applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp

```diff
@@ -506,6 +506,53 @@ double ComputePerturbationLocalMachNumber(const Element& rElement, const Process
     return velocity_module / local_speed_of_sound;
 }
 
+template <int Dim, int NumNodes>
+double ComputeUpwindFactor(
+        const double localMachNumberSquared, 
+        const ProcessInfo& rCurrentProcessInfo)
+{
+    // Following Fully Simulataneous Coupling of the Full Potential Equation
+    //           and the Integral Boundary Layer Equations in Three Dimensions
+    //           by Brian Nishida (1996), Equation 2.13
+
+    // read free stream values
+    // default MACH_LIMIT - 0.94 and default UPWIND_FACTOR_CONSTANT - 1.0
+    const double critical_mach = rCurrentProcessInfo[MACH_LIMIT];
```

Comment:
I would also add the variable CRITICAL_MACH as you suggested
```suggestion
    const double critical_mach = rCurrentProcessInfo[CRITICAL_MACH];
```